### PR TITLE
Rebase from old SOAP to Checkout REST API

### PR DESCRIPTION
**Pull request by rikterbeek**
**Description**

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: test_test.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+const RECURRING_TYPE_ONECLICK = "sd";
+
+$recurringType = "sdf";
+
+if($recurringType !== RECURRING_TYPE_ONECLICK) {
+    echo "ier";
+} else {
+    echo 'else';
+}
```
Comment (Aleffio) [inline test_test.php]: please remove

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 2]
```diff
@@ -66,6 +66,360 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+    /**
+     * Create a 3D secure payment request
+     *
+     * @param $payment
+     * @return mixed
+     */
+    public function authorise3DPayment($payment)
+    {
+        // retrieve configurations
+        if (Mage::app()->getStore()->isAdmin()) {
```
Comment (Aleffio) [inline app/code/community/Adyen/Payment/Model/Api.php]: I don't think we can do authorise3d as admin/backend

Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Api.php]: no correct fixed

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 3]
```diff
@@ -66,6 +66,360 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+                "houseNumberOrName" => '',
+                "country" => "ZZ"
+            );
+
+            // Save the defaults for later to compare if anything has changed
+            $requestDelivery = $requestDeliveryDefaults;
+
+
+            if (!empty($shippingAddress->getStreet(1))) {
+                $requestBilling["street"] = $shippingAddress->getStreet(1);
```
[the remark this answers is not in the document]
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Api.php]: Thanks, fixed

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 4]
```diff
@@ -66,6 +66,360 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+                "houseNumberOrName" => '',
+                "country" => "ZZ"
+            );
+
+            // Save the defaults for later to compare if anything has changed
+            $requestBilling = $requestBillingDefaults;
+
+
+            if (!empty($billingAddress->getStreet(1))) {
+                $requestBilling["street"] = $shippingAddress->getStreet(1);
```
[the remark this answers is not in the document]
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Api.php]: Thanks, fixed

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 5]
```diff
@@ -66,6 +66,360 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+            $storeId = null;
+        }
+
+        // configurations
+        $order = $payment->getOrder();
+        $orderCurrencyCode = $order->getOrderCurrencyCode();
+        $incrementId = $order->getIncrementId();
+        $realOrderId = $order->getRealOrderId();
+        $customerId = Mage::helper('adyen/payment')->getShopperReference($order->getCustomerId(), $realOrderId);
+        $merchantAccount = Mage::helper('adyen')->getAdyenMerchantAccount($this->_paymentMethod, $storeId);
```
[the remark this answers is not in the document]
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Api.php]: Thanks, fixed

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php [hunk 6]
```diff
@@ -559,17 +535,88 @@ protected function _processResponse(
...
+                );
+                $errorMsg = $errorMsg->getErrorMessage();
+
+                $this->resetReservedOrderId();
+                Adyen_Payment_Exception::throwException($errorMsg);
+                break;
+            case "Authorised":
+                $this->_addStatusHistory($payment, $responseCode, $pspReference, $this->_getConfigData('order_status'));
+                break;
+            case '[capture-received]':
```
Comment (rkewlani) [inline app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: case has single quote here but in previous checks it is double quotes. Can we make it consistent ?

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php [hunk 7]
```diff
@@ -559,17 +535,88 @@ protected function _processResponse(
...
+            case '[cancel-received]':
+            case '[cancelOrRefund-received]':
+                $this->_addStatusHistory($payment, $responseCode, $pspReference, false, null, $originalPspReference);
+                break;
+            case "Error":
+                $this->resetReservedOrderId();
+                $errorMsg = Mage::helper('adyen')->__('System error, please try again later');
+                Adyen_Payment_Exception::throwException($errorMsg);
+                break;
+            default:
```
Comment (rkewlani) [inline app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: Are you considering PresentToShopper in this flow ?

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 8]
```diff
@@ -66,6 +66,353 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+            if ($payment->getAdditionalInformation("store_cc") == "" &&
+                $recurringType === "ONECLICK,RECURRING"
+            ) {
+                $request['enableRecurring'] = true;
+            } elseif ($payment->getAdditionalInformation("store_cc") == "1") {
+                if ($recurringType == "ONECLICK" || $recurringType == "ONECLICK,RECURRING") {
+                    $request['paymentMethod']['storeDetails'] = true;
+                }
+
+                if ($recurringType == "ONECLICK,RECURRING" || $recurringType == "RECURRING") {
```
Comment (rkewlani) [inline app/code/community/Adyen/Payment/Model/Api.php]: Same here about quotes.

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php (hunk 7 above)
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: This is now moved to checkout.

File: app/code/community/Adyen/Payment/Model/Adyen/Abstract.php (hunk 6 above)
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Adyen/Abstract.php]: yes will make it consistent

File: app/code/community/Adyen/Payment/Model/Api.php (hunk 8 above)
Comment (rikterbeek) [reply app/code/community/Adyen/Payment/Model/Api.php]: agree.

**Pull request merged**